### PR TITLE
Add new Draw.io diagrams

### LINKED ISSUE #263: Command Groups - Update Composition Diagram to Clarify Behavior of "ParallelCommandGroup"
[Duplicated from artf5901 on the beta test site]

On the 2020 Online Documentation web page for Command Groups at https://frc-docs.readthedocs.io/en/latest/docs/software/commandbased/command-groups.html#recursive-composition-of-command-groups the diagram depicting the control flow of the "command group with concurrency" is a bit misleading.  In the example, there is a SequentialCommandGroup of DriveToGoal(), ParallelCommandGroup(RaiseElevator(), SetWristPosition()), and ScoreTube().  The diagram (as of the time of this tracker item) implies that the ScoreTube() command commences after the RaiseElevator() command finishes, regardless of the status of the SetWristPosition() command.    The diagram should be clarified to indicate that the ScoreTube() command does not commence until both the RaiseElevator() and SetWristPosition() commands have finished.  

I have attached a couple images which depict (a) the referenced image at the time of this writing, and (b) an updated image showing what the diagram could be clarified to look like.  (Both are attached.)  Obviously, the updated image would look better than what I have provided if the original editing tool to generate the image could be used instead of my crude markup of the original image in MS Paint.

image

image

(Incidentally, the original image is a reasonable depiction of the control flow if the composition were ParallelDeadlineGroup(RaiseElevator(), SetWristPosition()), but I wouldn't recommend using the DeadlineGroup example as the basic example of parallel composition.)

### LINKED ISSUE #622: Odd Diagrams in Git Version Control Introduction
The same diagram is currently shown in the [Branches section](https://docs.wpilib.org/en/latest/docs/software/basic-programming/git-getting-started.html#branches) and [Merging section](https://docs.wpilib.org/en/latest/docs/software/basic-programming/git-getting-started.html#merging). Is this intentional? What should be show differently (if anything) in the second diagram?

---

image

### PR DESCRIPTION
**Pull request by yellowjaguar5**
Create diagrams specified in #221 with draw.io. Please feel free to comment with any additional diagrams that should be created.

Closes #587, fixes #263, fixes #622 and closes #221.

Checklist (continued from #221):

- [x]  [Types of Sensors](https://frc-docs.readthedocs.io/en/latest/docs/software/sensors/sensor-overview-software.html)
- [x]  [Vision Workflows](https://frc-docs.readthedocs.io/en/latest/docs/software/vision-processing/introduction/strategies-for-vision-programming.html)
- [x]  [Sources and Sinks](https://frc-docs.readthedocs.io/en/latest/docs/software/vision-processing/introduction/cameraserver-class.html)
- [x]  [Subsystems and Commands](https://frc-docs.readthedocs.io/en/latest/docs/software/commandbased/what-is-command-based.html#id2)
- [x]  [Scheduler Run Sequence (2nd diagram on page)](https://frc-docs.readthedocs.io/en/latest/docs/software/commandbased/command-scheduler.html)
- [x]  [Command Group with Concurrency](https://frc-docs.readthedocs.io/en/latest/docs/software/commandbased/command-groups.html#id4)
- [x]  [NAT](https://frc-docs.readthedocs.io/en/latest/docs/networking/networking-introduction/networking-basics.html#public-vs-private-ip-addresses)

Diagrams from #191 have also been re-created in draw.io.

Potential diagrams to redo:

- X /software/actuators/images/axisconventions.jpg
- X /software/actuators/images/diffdrive.jpg
- [x] /networking/networking-introduction/images/networking-basics/mixing-static-dynamic.png
- X /software/sensors/images

Old command-based docs:

/software/old-commandbased/commands/images/scheduling-commands/image2.png
/software/old-commandbased/commands/images/scheduling-commands/image3.png
/software/old-commandbased/basics/images/what-is-commandbased/image2.png
/software/old-commandbased/basics/images/what-is-commandbased/image4.png
/software/old-commandbased/basics/images/what-is-commandbased/image5.png
/software/old-commandbased/basics/images/what-is-commandbased/image6.png
/software/old-commandbased/commands/images/synchronizing-two-commands/image2.png
/software/old-commandbased/commands/images/synchronizing-two-commands/image1.png

### CODE REVIEW

**Comment by ItayZiv** [comment]
Probably a bad idea to have the full commits of the PR to add draw.io support in here (Just make sure you remove it once PR is merged ig)?
Other than that, I have the subsystems and commands diagram and the parallel command group diagram in draw.io (including a fix to the latter since its actually quite inaccurate).

**Comment by Daltz333** [comment]
Doesn't build on RTD as it doesn't have sudo.

**Comment by yellowjaguar5** [comment]
@Daltz333 I removed sudo, let me know if that fixes the issue.

**Comment by ItayZiv** [comment]
Just a btw, this still relies on https://github.com/wpilibsuite/frc-docs/pull/602, which isn’t done because style-guide changes aren’t complete neither is the extension itself (Were getting close).
Another note, if you notice that right now text in diagrams may look bad / hard to read when converted, that’s one thing that’s basically fixed (unit testing not done yet) in the extension.

**Comment by ItayZiv** [comment]
Ok so as far as I know, currently draw.io support won't be happening in RTD, since installing packages in RTD environment isn't really possible, from what I understand, currently, you should just export all the diagrams to SVG from draw.io.
I would recommend keeping a hand on the original files, just in case they are ever needed.
(Will update if something I said here is incorrect)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -32,6 +32,14 @@ jobs:
     - script: sudo apt-get install -qy --force-yes graphviz
       displayName: 'Install GraphViz'
 
+    - script: |
```
Comment (Daltz333) [inline azure-pipelines.yml:35]: No need for the azure changes

File: source/conf.py [hunk 2]
```diff
@@ -13,7 +13,26 @@
 # import os
 # import sys
 # sys.path.insert(0, os.path.abspath('.'))
-
+import os
```
Comment (Daltz333) [inline source/conf.py:16]: Remove the conf.py changes

File: source/docs/software/vision-processing/introduction/target-info-and-retroreflection.rst [hunk 3]
```diff
@@ -22,11 +22,14 @@ Retro-reflective materials are not mirrored, but it will typically have either s
 Examples of Retroreflection
 ---------------------------
 
-.. image:: images/target-info-and-retroreflection/examples-of-retroreflection.png
+.. image:: images/target-info-and-retroreflection/raised-pavement-marker.jpg
+  :width: 49%
```
Comment (Daltz333) [inline source/docs/software/vision-processing/introduction/target-info-and-retroreflection.rst:26]: Don't use percentages.

Comment (yellowjaguar5) [reply source/docs/software/vision-processing/introduction/target-info-and-retroreflection.rst]: Fixed in 92eb5b4.

**Comment by ItayZiv** [comment]
Basically all the commits related to the draw.io extension (should be all commits I did here), are outdated and need to be reverted (or rebased)

File: azure-pipelines.yml (hunk 1 above)
Comment (yellowjaguar5) [reply azure-pipelines.yml]: Fixed in 4cf0ed9

File: source/conf.py (hunk 2 above)
Comment (yellowjaguar5) [reply source/conf.py]: Fixed in 4cf0ed9.

**Comment by yellowjaguar5** [comment]
@ItayZiv I believe I've reversed all the changes made in your commits to this PR. When this PR gets merged, it will be squashed, so the commits won't end up in the master history. Anything I'm missing?

**Comment by yellowjaguar5** [comment]
I removed the sensor overview diagram per a conversation with @Daltz333

**Pull request merged**